src/main/java/appeng/client/gui/implementations/GuiFluidTerminal.java

```diff
@@ -0,0 +1,362 @@
+		this.fontRenderer.drawString( GuiText.inventory.getLocal(), 8, this.ySize - 96 + 3, 4210752 );
+	}
+
+	@Override
+	public void drawBG( int offsetX, int offsetY, int mouseX, int mouseY )
+	{
+		this.bindTexture( this.getBackground() );
+		final int x_width = 197;
+		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, x_width, 18 );
+
+		for( int x = 0; x < 6; x++ )
+		{
+			this.drawTexturedModalRect( offsetX, offsetY + 18 + x * 18, 0, 18, x_width, 18 );
+		}
+
+		this.drawTexturedModalRect( offsetX, offsetY + 16 + 6 * 18, 0, 106 - 18 - 18, x_width, 99 + 77 );
+
+		if( this.searchField != null )
+		{
+			this.searchField.drawTextBox();
+		}
+	}
+
+	@Override
+	public void updateScreen()
+	{
+		this.repo.setPower( this.container.isPowered() );
+		super.updateScreen();
+	}
+
+	@Override
+	protected void renderHoveredToolTip( int mouseX, int mouseY )
+	{
+		Slot slot = this.getSlot( mouseX, mouseY );
+		if( slot != null && slot.isEnabled() && slot instanceof ISlotFluid )
+		{
+			ISlotFluid fluidSlot = (ISlotFluid) slot;
+			if( fluidSlot.getFluidInSlot() != null && fluidSlot.shouldRenderAsFluid() )
+			{
+				FluidStack fluidStack = fluidSlot.getFluidInSlot();
+				List<String> list = new ArrayList<>();
+				list.add( fluidStack.getLocalizedName() );
+				int amount = fluidStack.amount;
+				DecimalFormat formatter = new DecimalFormat( "#,###.##" );
+				String amountStr = formatter.format( amount ) + "mb";
+				if( amount >= 10000 )
+				{
+					amountStr += " ( ";
+					amountStr += formatter.format( amount / 1000 ) + "B";
+					amountStr += " )";
+				}
+				list.add( amountStr );
+				String modName = "";
+				modName += TextFormatting.BLUE;
+				modName += TextFormatting.ITALIC;
+				modName += Loader.instance().getIndexedModList().get( Platform.getModId( AEFluidStack.fromFluidStack( fluidStack ) ) ).getName();
+				list.add( modName );
+				this.drawHoveringText( list, mouseX, mouseY );
+				return;
+			}
+		}
+		super.renderHoveredToolTip( mouseX, mouseY );
+	}
+
+	@Override
+	protected void actionPerformed( GuiButton btn ) throws IOException
+	{
+		if( btn instanceof GuiImgButton )
+		{
+			final boolean backwards = Mouse.isButtonDown( 1 );
+
+			final GuiImgButton iBtn = (GuiImgButton) btn;
+			if( iBtn.getSetting() != Settings.ACTIONS )
+			{
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				try
+				{
+					NetworkHandler.instance().sendToServer( new PacketValueConfig( iBtn.getSetting().name(), next.name() ) );
+				}
+				catch( final IOException e )
+				{
+					AELog.debug( e );
+				}
+
+				iBtn.set( next );
+			}
+		}
+	}
+
+	@Override
+	protected void handleMouseClick( Slot slot, int slotIdx, int mouseButton, ClickType clickType )
+	{
+		if( slot instanceof SlotFluidME )
+		{
+			SlotFluidME meSlot = (SlotFluidME) slot;
+			if( clickType == ClickType.PICKUP )
+			{
+				// TODO: Allow more options

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)